email/blocker.go

```diff
@@ -85,83 +83,92 @@ func (b *Blocker) threadedBlockMessages() {
 	abuseDB := b.staticDatabase
 	logger := b.staticLogger
 
+	// create a new ticker
 	ticker := time.NewTicker(blockFrequency)
-	first := true
 
 	// start the loop
 	for {
-		// sleep until next iteration, sleeping at the start of the for loop
-		// allows to 'continue' on error.
-		if !first {
-			select {
-			case <-b.staticContext.Done():
-				logger.Debugln("Blocker context done")
+		logger.Debugln("Triggered")
+
+		func() {
```

Comment:
Seems like a perfect candidate to be pulled out into another function.

Reply:
Done